lib/SoftwareTree.py

```diff
@@ -1,301 +1,213 @@
 import sublime_plugin, sublime
+from copy import deepcopy
+from threading import Thread, Timer, Event 
+from .SoftwareOffline import getSessionSummaryData, launchCodeTimeMetrics
+from .SoftwareUtil import *
+from .SoftwareModels import SessionSummary
+from .SoftwareWallClock import *
+from .SoftwareFileChangeInfoSummaryData import *
+from .SoftwareRepo import *
 
-class OpenTreeView(sublime_plugin.TextCommand):
+icons = getIcons()
+tree_view = None
+orig_layout = None
+NO_ID = 'NO_ID'
+CODETIME_TREEVIEW_NAME = 'Code Time Tree View'
+shouldOpen = True
 
-    def run(self, edit):
-        tree_view = None
-        window = self.view.window()
+def setShouldOpen(val):
+    global shouldOpen
+    shouldOpen = val
+
+'''
+Because we do frequent tree updates/refreshes, we don't want to redraw the tree in its unopened
+state, especially if the user is in the middle of using it. This data struct keeps track of what
+was opened so state can be maintained across draws.
+'''
+open_state = set()
+
+# IDs for code time action buttons
+CODE_TIME_ACTIONS = {'advanced-metrics', 'generate-dashboard', 'toggle-status-metrics', 'learn-more', 'submit-feedback'}
+
+# TODO: rare bug where tree isn't clickable (possibly slow wifi)
+class OpenTreeView(sublime_plugin.WindowCommand):
+
+    def run(self):
+        if not shouldOpen:
+            return
+
+        global tree_view 
+        global orig_layout
+        self.currentKeystrokeStats = SessionSummary()
+        window = self.window
         orig_view = window.active_view()
-        if not tree_view:
+        
+        if tree_view is None:
```

Comment:
Could move this logic to build the tree into another class method called `render_tree` to split up some of the code